core/src/main/scala/org/apache/spark/deploy/worker/Worker.scala

```diff
@@ -488,6 +499,16 @@ private[deploy] class Worker(
       logInfo(s"Master with url $masterUrl requested this worker to reconnect.")
       registerWithMaster()
 
+    case MasterInRevoking(masterUrl) =>
+      logWarning(s"Master with url $masterUrl is being revoked, current active" +
+        s" masterUrl $activeMasterUrl.")
+      if (masterUrl == activeMasterUrl) {
+        registerWithMaster()
```

Comment:
I'm hesitate about this part. Do we really need to `registerWithMaster()` when `masterUrl == activeMasterUrl` ? `MasterChanged` haven't come at this time, but it will comes soon because Master has been re-elected. What's more, it is highly possible that we'll hit *duplicate worker registration* since the new leader master is in recovery.

Maybe, we should process this message just like `MasterInStandby` do ? 

We need to think more about it.